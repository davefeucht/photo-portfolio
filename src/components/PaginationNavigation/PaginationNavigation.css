.pagination-navigation {
    width: 100%;
    height: 40px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    background-color: var(--template-black);
    grid-column: 1 / -1;
}