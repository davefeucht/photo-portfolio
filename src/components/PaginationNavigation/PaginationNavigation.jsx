/** **************
* PaginationNavigation component displays the navigation for pagination of items
*************** */

import './PaginationNavigation.css';

import PropTypes from 'prop-types';
import React from 'react';

import PaginationNavigationItem from '../PaginationNavigationItem/PaginationNavigationItem.jsx';

const prepNavigationItems = (totalPages, currentPageIndex) => {
    const navigationItems = [];
    for (let i = 0; i < totalPages; i++) {
        navigationItems.push({
            content: `${i + 1}`,
            selectedState: i === (currentPageIndex - 1) ? 'selected' : 'deselected'
        });
    }

    return navigationItems;
};

const PaginationNavigation = ({ totalPages, currentPageIndex, navigationFunction }) => {
    return (
        <div className="pagination-navigation">
            <div className="pagination-navigation__content">
                {prepNavigationItems(totalPages, currentPageIndex).map((item, index, array) => {
                    return (
                        <PaginationNavigationItem
                            key={`pagination_navigation_${item.content}`}
                            content={item.content}
                            selectedState={item.selectedState}
                            navigationFunction={navigationFunction}
                            isLast={index === array.length - 1}
                        />
                    );
                })}
            </div>
        </div>
    );
};

PaginationNavigation.displayName = 'PaginationNavigation';

PaginationNavigation.propTypes = {
    totalPages: PropTypes.number.isRequired,
    currentPageIndex: PropTypes.number.isRequired,
    navigationFunction: PropTypes.func.isRequired
};

export default PaginationNavigation;
