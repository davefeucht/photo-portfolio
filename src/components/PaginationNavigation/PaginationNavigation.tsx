/** **************
* PaginationNavigation component displays the navigation for pagination of items
*************** */

import './PaginationNavigation.css';

import * as React from 'react';

import PaginationNavigationItem from '../PaginationNavigationItem/PaginationNavigationItem';

interface PaginationNavigationProps {
    totalPages: number,
    currentPageIndex: number,
    navigationFunction(content: number): void
}

const prepNavigationItems = (totalPages: number, currentPageIndex: number) => {
    const navigationItems = [];
    for (let i = 0; i < totalPages; i++) {
        navigationItems.push({
            content: `${i + 1}`,
            selectedState: i === (currentPageIndex - 1) ? 'selected' : 'deselected'
        });
    }

    return navigationItems;
};

const PaginationNavigation: React.FC<PaginationNavigationProps> = ({ totalPages, currentPageIndex, navigationFunction }) => {
    return (
        <div className="pagination-navigation">
            <div className="pagination-navigation__content">
                {prepNavigationItems(totalPages, currentPageIndex).map((item, index, array) => {
                    return (
                        <PaginationNavigationItem
                            key={`pagination_navigation_${item.content}`}
                            content={item.content}
                            selectedState={item.selectedState}
                            navigationFunction={navigationFunction}
                            isLast={index === array.length - 1}
                        />
                    );
                })}
            </div>
        </div>
    );
};

PaginationNavigation.displayName = 'PaginationNavigation';

export default PaginationNavigation;
