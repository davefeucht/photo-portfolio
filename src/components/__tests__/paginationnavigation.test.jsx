import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import PaginationNavigation from '../PaginationNavigation/PaginationNavigation';

const totalPages = 14;
const currentPageIndex = 1;
const navigationFunction = () => {};

test('PaginationNavigation displays', async () => {
    const component = renderer.create(
        <MemoryRouter>
            <PaginationNavigation totalPages={totalPages} currentPageIndex={currentPageIndex} navigationFunction={navigationFunction} />
        </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
