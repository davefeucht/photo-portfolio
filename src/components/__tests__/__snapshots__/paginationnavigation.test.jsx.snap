// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNavigation displays 1`] = `
<div
  class="pagination-navigation"
>
  <div
    class="pagination-navigation__content"
  >
    <div
      aria-label="pagination"
      class="pagination-navigation__item selected"
    >
      1
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      2
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      3
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      4
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      5
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      6
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      7
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      8
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      9
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      10
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      11
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      12
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      13
       | 
    </div>
    <div
      aria-label="pagination"
      class="pagination-navigation__item deselected"
    >
      14
    </div>
  </div>
</div>
`;
